third_party/terraform/tests/resource_google_billing_subaccount_test.go

```diff
@@ -0,0 +1,141 @@
+package google
+
+import (
+	"fmt"
+	"strings"
+	"testing"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccBillingSubaccount_renameOnDestroy(t *testing.T) {
+	t.Parallel()
+
+	masterBilling := getTestMasterBillingAccountFromEnv(t)
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckGoogleBillingSubaccountRenameOnDestroy,
+		Steps: []resource.TestStep{
+			{
+				// Test Billing Subaccount creation
+				Config: testAccBillingSubccount_renameOnDestroy(masterBilling),
+				Check:  testAccCheckGoogleBillingSubaccountExists("subaccount_with_rename_on_destroy"),
+			},
+		},
+	})
+}
+
+func TestAccBillingSubaccount_basic(t *testing.T) {
+	t.Parallel()
+
+	masterBilling := getTestMasterBillingAccountFromEnv(t)
+	resource.Test(t, resource.TestCase{
+		PreCheck:  func() { testAccPreCheck(t) },
+		Providers: testAccProviders,
+		Steps: []resource.TestStep{
+			{
+				// Test Billing Subaccount creation
+				Config: testAccBillingSubccount_basic(masterBilling),
+				Check:  testAccCheckGoogleBillingSubaccountExists("subaccount"),
+			},
+			{
+				ResourceName:            "google_billing_subaccount.subaccount",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"deletion_policy"},
+			},
+			{
+				// Test Billing Subaccount update
+				Config: testAccBillingSubccount_update(masterBilling),
+				Check:  testAccCheckGoogleBillingSubaccountExists("subaccount"),
+			},
+			{
+				ResourceName:            "google_billing_subaccount.subaccount",
+				ImportState:             true,
+				ImportStateVerify:       true,
+				ImportStateVerifyIgnore: []string{"deletion_policy"},
+			},
+		},
+	})
+}
+
+func testAccBillingSubccount_basic(masterBillingAccountId string) string {
+	return fmt.Sprintf(`
+resource "google_billing_subaccount" "subaccount" {
+  display_name = "Test Billing Subaccount"
+  master_billing_account  = "%s"
+}
+`, masterBillingAccountId)
+}
+
+func testAccBillingSubccount_update(masterBillingAccountId string) string {
+	return fmt.Sprintf(`
+resource "google_billing_subaccount" "subaccount" {
+  display_name = "Rename Test Billing Subaccount"
+  master_billing_account  = "%s"
+}
+`, masterBillingAccountId)
+}
+
+func testAccBillingSubccount_renameOnDestroy(masterBillingAccountId string) string {
+	return fmt.Sprintf(`
+resource "google_billing_subaccount" "subaccount_with_rename_on_destroy" {
+  display_name = "Test Billing Subaccount (Rename on Destroy)"
+  master_billing_account  = "%s"
+  deletion_policy = "RENAME_ON_DESTROY"
+}
+`, masterBillingAccountId)
+}
+
+func testAccBillingSubccount_updateRenameOnDestroy(masterBillingAccountId string) string {
```

Comment:
This is unused, and can be removed.

Reply:
removed testAccBillingSubccount_updateRenameOnDestroy